lib/src/native/realm_core.dart

```diff
@@ -181,6 +190,115 @@ class _RealmCore {
     });
   }
 
+  RealmHttpTransportHandle createHttpTransport(HttpClient httpClient) {
+    return RealmHttpTransportHandle._(_realmLib.realm_dart_http_transport_new(Pointer.fromFunction(request_callback), httpClient));
+  }
+
+  static void request_callback(Object userData, realm_http_request request, Pointer<Void> request_context) {
+    //
+    // The request struct only survives until end-of-call, even though
+    // we explicitly call realm_http_transport_complete_request to
+    // mark request as completed later.
+    //
+    // Therefor we need to copy everything out of request before returning.
+    // We cannot clone request on the native side with realm_clone,
+    // since realm_http_request does not inherit from WrapC.
+    //
+    final client = userData as HttpClient;
+    client.connectionTimeout = Duration(milliseconds: request.timeout_ms);
+
+    final url = Uri.parse(request.url.cast<Utf8>().toDartString());
+
+    final method = _HttpMethod.values[request.method];
+
+    final body = request.body.cast<Utf8>().toDartString();
+
+    final headers = <String, String>{};
+    for (int i = 0; i < request.num_headers; ++i) {
+      final header = request.headers[i];
+      final name = header.name.cast<Utf8>().toDartString();
+      final value = header.value.cast<Utf8>().toDartString();
+      headers[name] = value;
+    }
+
+    _request_callback_async(client, method, url, body, headers, request_context);
+    // The request struct dies here!
+  }
+
+  static void _request_callback_async(
+    HttpClient client,
+    _HttpMethod method,
+    Uri url,
+    String body,
+    Map<String, String> headers,
+    Pointer<Void> request_context,
+  ) async {
+    await using((arena) async {
+      final response_pointer = arena<realm_http_response>();
+      final responseRef = response_pointer.ref;
+      // pre-fill in case something fails
+      responseRef.custom_status_code = 100; // TODO!
+
+      try {
+        // Build request
+        late HttpClientRequest request;
+        switch (method) {
+          case _HttpMethod.delete:
+            request = await client.deleteUrl(url);
+            break;
+          case _HttpMethod.put:
+            request = await client.putUrl(url);
+            break;
+          case _HttpMethod.patch:
+            request = await client.patchUrl(url);
+            break;
+          case _HttpMethod.post:
+            request = await client.postUrl(url);
+            break;
+          case _HttpMethod.get:
+            request = await client.getUrl(url);
+            break;
```

Comment:
We cover all cases. If a new case appear the compiler will warn

Reply:
hm then I guess my question is how are we sure that the conversion on line 212 works. This could be some other http method which we don't have in the enum. Then it will fail on that line. Could we catch this error and throw more meaningful one like "Http method not supported"

Reply:
my suggestion was to `switch` on the incoming number and `case` with the enum. Then we could have a default that throws the "http method not supported". But as written it will also work we just need to handle the conversion error above